astropy/constants/constant.py

```diff
@@ -159,10 +159,10 @@ def name(self):
         return self._name
 
     @lazyproperty
-    def unit(self):
+    def _unit(self):
```

Comment:
Maybe I am the one who misunderstood: `Constants` are intialised with units passed in as a string. My impression was that this unit could not be parsed immediately since some units depend on constants. Is this true? I thought this was the reason they were passed through `Unit` only later, using `lazyproperty`. The change I made was to rename the string-based unit from `_unit` to `_unit_string`, so that both `_unit` and `unit` can return `Unit` objects (since it was a problem for `Quantity` if `_unit` was a string rather than a `Unit` instance).